tests/src/test/scala/cats/tests/FoldableTests.scala

```diff
@@ -204,6 +204,28 @@ class FoldableTestsAdditional extends CatsSuite {
     // test laziness of foldM
     dangerous.foldM(0)((acc, a) => if (a < 2) Some(acc + a) else None) should === (None)
   }
+
+  test(".foldLeftM short-circuiting") {
+    val ns = Stream.continually(1)
+    val res = Foldable[Stream].foldLeftM[Either[Int, ?], Int, Int](ns, 0) { (sum, n) =>
+      if (sum >= 100000) Left(sum) else Right(sum + n)
+    }
+    assert(res == Left(100000))
+  }
+
+  test(".foldLeftM short-circuiting optimality") {
```

Comment:
This test fails if we remove `Foldable.iteratorFoldM`, because it actually no longer tests the default `foldM` implementation from `Foldable`  after #1414 which overrides foldLeftM in the instance of `Stream` using `iteratorFoldM`. This test fails when using the default `foldM` because, if I am not mistaken, the new `foldM` still have to evaluate one element after the stop. Also these two tests are now redundant with the test("Foldable[Stream]") above. I will fix this in a PR removing `iteratorFoldM`